packager/util/util.go

```diff
@@ -75,6 +76,17 @@ func hasCanSignHttpExchangesExtension(cert *x509.Certificate) bool {
 	return false
 }
 
+// Returns the Duration of time before cert expires with given deadline.
+// Returns -1 if cert is already expired. This will be used to periodically check if cert
+// is still within validity range.
+func GetDurationToExpiry(cert *x509.Certificate, currentTime time.Time) time.Duration {
+	if cert.NotBefore.After(currentTime) || cert.NotAfter.Before(currentTime) {
```

Comment:
Depending on how this ends up being used, I'd suggest two changes:
1. Drop the `NotAfter.Before(currentTime)` bit; the return statement below will already return a negative value in that case.
2. Print a warning to the screen when `NotBefore.After(currentTime)`; this probably indicates some configuration error (either with the CA, or the local clock), and it would be good to have some knowledge if the server ends up spinning in a loop because of this.

If you choose to do #2 in the caller function instead of here (reasonable choice either way), then I'd prefer returning (time.Duration, error) instead of using a sentinel value.

Reply:
I decided to keep #1 because the logic seems cleaner if I can just return a valid duration plus nil (for the error).
I also decided to let the caller handle the error message.

Reply:
I think that it is no use to check if the cert is expired at the `currentTime` because the cert should be valid until the SXG signature is expired. It is too late to know its expiry at the `currentTime`.
The signature expiry time of `Expires:     now.Add(6 * 24 * time.Hour)` needs to be considered and it is better to add it in this check.

Reply:
I agree, this utility is not meant to check it at current time (as in time.Now()), it was meant to check the cert against a specified deadline.  Your point about the naming of the parameters is well taken, I will rename it to certExpiryDeadline.

Reply:
Maybe add a note, too, to the function doc that this should be the expected SXG expiration time.

@shigeki Also note that we discussed making a toml config parameter (something like renewalGracePeriod), with a default of say, 2 days, so that the process for requesting a new cert via ACME will begin 6+2=8 days before expiration, to allow for ACME server outages. Does this seem OK?

Reply:
c365be83b9490e95ba4b46de67c58b0b294da0d3 is fine for me. Thanks.